demo/xcb_cairo/nuklear_xcb.h

```diff
@@ -0,0 +1,852 @@
+
+#include <xcb/xcb.h>
+#include <xcb/xcb_util.h>
+#include <xcb/xcb_keysyms.h>
+#include <X11/keysym.h>
+#include <cairo/cairo-xcb.h>
+#include <cairo/cairo-ft.h>
+
+#if defined _XOPEN_SOURCE && _XOPEN_SOURCE >= 600 || \
+    defined _POSIX_C_SOURCE && _POSIX_C_SOURCE >= 200112L
+#include <time.h>
+#include <errno.h>
+#ifndef NK_XCB_FPS
+#define NK_XCB_FPS 30
+#endif /* NK_XCB_FPS */
+#define NK_XCB_NSEC 1000000000
+#define NK_XCB_MIN_FRAME_TIME (NK_XCB_NSEC / NK_XCB_FPS)
+#endif
+
+#include <math.h>
+#ifdef __USE_GNU
+#define NK_XCB_PI M_PIl
+#elif defined __USE_BSD || defined __USE_XOPEN
+#define NK_XCB_PI M_PI
+#else
+#define NK_XCB_PI acos(-1.0)
+#endif
+
+#define NK_XCB_TO_CAIRO(x) ((double) x / 255.0)
+#define NK_XCB_DEG_TO_RAD(x) ((double) x * NK_XCB_PI / 180.0)
+
+struct nk_cairo_context {
+    cairo_surface_t *surface;
+    cairo_t *cr;
+
+    struct nk_user_font *font;
+    struct nk_color *bg;
+
+    void *last_buffer;
+    nk_size buffer_size;
+    int repaint;
+};
+
+struct nk_xcb_context {
+    xcb_connection_t *conn;
+    int screennum;
+    xcb_window_t window;
+    xcb_key_symbols_t *key_symbols;
+#ifdef NK_XCB_MIN_FRAME_TIME
+    unsigned long last_render;
+#endif /* NK_XCB_MIN_FRAME_TIME */
+    int events;
+    xcb_intern_atom_reply_t* del_atom;
+    int width, height;
+};
+
+NK_API struct nk_xcb_context *nk_xcb_init(const char *title, int pos_x, int pos_y, int width, int height)
+{
+    int screenNum;
+    xcb_connection_t *conn;
+    xcb_screen_t *screen;
+    xcb_window_t window;
+    uint32_t values[1];
+    struct nk_xcb_context *xcb_ctx;
+    xcb_intern_atom_cookie_t cookie;
+    xcb_intern_atom_reply_t *reply, *del_atom;
+
+    conn = xcb_connect(NULL, &screenNum);
+    if (xcb_connection_has_error(conn)) {
+        xcb_disconnect(conn);
+        return NULL;
+    }
+    screen = xcb_aux_get_screen(conn, screenNum);
+
+    window = xcb_generate_id(conn);
+    values[0] = XCB_EVENT_MASK_KEY_PRESS
+        | XCB_EVENT_MASK_KEY_RELEASE
+        | XCB_EVENT_MASK_BUTTON_PRESS
+        | XCB_EVENT_MASK_BUTTON_RELEASE
+        /*| XCB_EVENT_MASK_ENTER_WINDOW*/
+        /*| XCB_EVENT_MASK_LEAVE_WINDOW*/
+        | XCB_EVENT_MASK_POINTER_MOTION
+        /*| XCB_EVENT_MASK_POINTER_MOTION_HINT*/
+        | XCB_EVENT_MASK_BUTTON_1_MOTION
+        | XCB_EVENT_MASK_BUTTON_2_MOTION
+        | XCB_EVENT_MASK_BUTTON_3_MOTION
+        | XCB_EVENT_MASK_BUTTON_4_MOTION
+        | XCB_EVENT_MASK_BUTTON_5_MOTION
+        | XCB_EVENT_MASK_BUTTON_MOTION
+        | XCB_EVENT_MASK_KEYMAP_STATE
+        | XCB_EVENT_MASK_EXPOSURE
+        /*| XCB_EVENT_MASK_VISIBILITY_CHANGE*/
+        | XCB_EVENT_MASK_STRUCTURE_NOTIFY
+        /*| XCB_EVENT_MASK_RESIZE_REDIRECT*/
+        /*| XCB_EVENT_MASK_SUBSTRUCTURE_NOTIFY*/
+        /*| XCB_EVENT_MASK_SUBSTRUCTURE_REDIRECT*/
+        /*| XCB_EVENT_MASK_FOCUS_CHANGE*/
+        /*| XCB_EVENT_MASK_PROPERTY_CHANGE*/
+        /*| XCB_EVENT_MASK_COLOR_MAP_CHANGE*/
+        /*| XCB_EVENT_MASK_OWNER_GRAB_BUTTON*/
```

Comment:
I'm guessing the commented out events are to show what is available, but the demo doesn't need it?